packages/pg-typed/src/index.ts

```diff
@@ -290,15 +290,35 @@ class Table<TRecord, TInsertParameters> {
   }
 
   async insert<TRecordsToInsert extends readonly TInsertParameters[]>(
-    ...rows: TRecordsToInsert
-  ): Promise<{[key in keyof TRecordsToInsert]: TRecord}> {
+    ...rows: keyof TRecordsToInsert[number] extends keyof TInsertParameters
```

Comment:
These type changes could use some types unit tests to guarantee unwanted regressions.